src/overview/main.d

```diff
@@ -1,8 +1,91 @@
+/*******************************************************************************
+
+    Copyright:
+        Copyright (c) 2009-2016 Sociomantic Labs GmbH. All rights reserved.
+
+    License:
+        Boost Software License Version 1.0. See LICENSE.txt for details.
+
+*******************************************************************************/
+
 module overview.main;
 
 import octod.core;
 import octod.api.repos;
+import overview.repository;
 
 void main ( )
 {
+    import overview.htmlreport;
+    import semver.Version;
+    import vibe.core.log : logInfo;
+    import std.algorithm.iteration : map, filter, each;
+    import std.range : array;
+
+    Configuration conf;
+    conf.dryRun = false;
+    conf.oauthToken = "READ-ME-FROM-CONFIG";
+    auto client = HTTPConnection.connect(conf);
+
+    logInfo("Fetching repository list");
+    auto repos = client.fetchAllProjects();
+
+    logInfo("Aggregating initial repository metadata");
+    auto projects = repos
+        .map!(name => client.fetchRepositoryMetadata(name))
+        .array();
+
+    logInfo("Building SHA to version identifier mapping");
+    Version[string] sha_mapping;
+    projects
+        .filter!(proj => !proj.library)
+        .map!(proj => proj.tags)
+        .each!(aa => aa.mergeInto(sha_mapping));
+
+    logInfo("Resolving dependency versions for all projects");
+    foreach (ref project; projects)
+        project.updateRepositoryDependencies(client, sha_mapping);
+
+    logInfo("Generating HTML report");
+    generateHTMLReport(projects, "./report.html");
+}
+
+/**
+    Uses GitHub API to get all sociomantic D projects excluding some
+    pre-defined list
+ **/
+string[] fetchAllProjects ( HTTPConnection client )
+{
+    static immutable excluded_projects = [ "ocean", "tango", "phobos",
+        "chest", "druntime", "dtools" ];
+
+    import std.algorithm.searching : canFind;
+    import std.algorithm.iteration : filter, map;
+    import std.array : array;
+    import std.exception : ifThrown;
+
+    auto repos = client
+        .listOrganizationRepos("sociomantic")
```

Comment:
I think this should be moved to a command-line argument / config file. It makes no sense to publish a script that has an internal / private organization hardcoded (same for exclude projects, etc.)